assets/js/blocks/products/all-products/edit.tsx

```diff
@@ -48,51 +47,58 @@ import { getSharedContentControls, getSharedListControls } from '../edit';
 import Block from './block';
 import './editor.scss';
 
-/**
- * Component to handle edit mode of "All Products".
- */
-class Editor extends Component {
-	static propTypes = {
-		/**
-		 * The attributes for this block.
-		 */
-		attributes: PropTypes.object.isRequired,
-		/**
-		 * A callback to update attributes.
-		 */
-		setAttributes: PropTypes.func.isRequired,
-		/**
-		 * From withSpokenMessages.
-		 */
-		debouncedSpeak: PropTypes.func.isRequired,
-	};
+type Attributes = {
+	columns: number;
+	rows: number;
+	alignButtons: boolean;
+	layoutConfig: [ string, object? ][];
+};
 
-	state = {
-		isEditing: false,
-		innerBlocks: [],
-	};
+type Props = {
+	contentVisibility: number;
+	orderby: number;
+	isPreview: number;
+	clientId: string;
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+};
 
-	blockMap = getBlockMap( 'woocommerce/all-products' );
+export default function Edit( props: Props ): JSX.Element {
+	const [ isEditing, setIsEditing ] = useState< boolean >( false );
+	const [ , setInnerBlocks ] = useState< BlockInstance[] | boolean >( false );
```

Comment:
To preserve the logic, I believe the initial state should be an empty array as per previous implementation:

```
	state = {
		isEditing: false,
		innerBlocks: [],
	};
```

Also, could you shed more light on why there's just a `setInnerBlocks` function in use, but no `innerBlocks` value like:
```suggestion
	const [ innerBlocks, setInnerBlocks ] = useState< BlockInstance[] | boolean >( false );
```